### PR TITLE
MiniAOD phase2 updates: updated PUPPI tune, removed old MET filters

### PR DESCRIPTION
**Pull request by gpetruc**
PUPPI tune taken from https://twiki.cern.ch/twiki/bin/view/CMS/UPGTrackerTDRStudies + some discussion with Phil to fix some odd ends.
Checked on a few H &rarr; &gamma;&gamma; events that the run 2 photon ID still makes some sense also at PU140 (i.e. the photons selected by PuppiPhoton are mostly the same between PU0, PU35 and PU140), so there is no need to switch it off.

@violatingcp @nhanvtran @kpedro88

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by kpedro88** [comment]
assign upgrade

**Comment by kpedro88** [comment]
+1
Comparison differences in 2023 workflows are concentrated in JetMET-related quantities, as expected, and the changes are minor.

**Comment by slava77** [comment]
@gpetruc 
I didn't find any puppi performance plots with this tune in https://twiki.cern.ch/twiki/bin/view/CMS/UPGTrackerTDRStudies
It is needed for PR documentation/approval.

**Comment by violatingcp** [comment]
Hi Slava,

See here :
https://indico.cern.ch/event/623893/contributions/2531745/attachments/1436313/2208833/PCH_Upgrade_29_3.pdf

I will attach it to the twiki now.

Phil

**Comment by slava77** [comment]
On 4/13/17 9:45 AM, Philip Harris wrote:
> Hi Slava,
> 
> See here :
> https://indico.cern.ch/event/623893/contributions/2531745/attachments/1436313/2208833/PCH_Upgrade_29_3.pdf

Hi Phil,

Thank you for the pointer to the slides.
It would be quite useful for this PR to know the comparison between the
current and the proposed versions.
I missed it in these slides.


		--slava

**Comment by violatingcp** [comment]
Hi Slava,

The big change is the extension of the eta range to include  forward
tracking. I can run with the old setup, but the change is well motivated
given the forward tracking.

Phil

**Comment by slava77** [comment]
Hi Phil

I will run on ~100 events (if relvals are easily accessible) and will post what's in the standard DQM validation.
If nothing odd shows up, then OK to merge this; otherwise I'll ask for more plots

Thanks

File: CommonTools/PileupAlgos/python/Puppi_cff.py [hunk 1]
```diff
@@ -80,3 +80,42 @@
...
+             puppiAlgos = puppiCentral
+        ), cms.PSet( 
+             etaMin = cms.vdouble(2.5),
+             etaMax = cms.vdouble(3.5),
+             ptMin  = cms.vdouble(0.),#Normally 0
+             MinNeutralPt   = cms.vdouble(0.2),
+             MinNeutralPtSlope   = cms.vdouble(0.030),
+             RMSEtaSF = cms.vdouble(1.0),
+             MedEtaSF = cms.vdouble(1.0),
+             EtaMaxExtrap = cms.double(2.0),
```
Comment (slava77) [inline CommonTools/PileupAlgos/python/Puppi_cff.py]: is this value correct? (the range is 2.5 to 3.5 here)

File: CommonTools/PileupAlgos/python/Puppi_cff.py [hunk 2]
```diff
@@ -80,3 +80,42 @@
...
+             etaMax = cms.vdouble(2.5),
+             ptMin  = cms.vdouble(0.),
+             MinNeutralPt   = cms.vdouble(0.2),
+             MinNeutralPtSlope   = cms.vdouble(0.015),
+             RMSEtaSF = cms.vdouble(1.0),
+             MedEtaSF = cms.vdouble(1.0),
+             EtaMaxExtrap = cms.double(2.0),
+             puppiAlgos = puppiCentral
+        ), cms.PSet( 
+             etaMin = cms.vdouble(2.5),
```
Comment (slava77) [inline CommonTools/PileupAlgos/python/Puppi_cff.py]: IIUC, the design is that the PSets with the same puppiAlgos should be merged into smth like
```python
             etaMin = cms.vdouble(0., 2.5),
             etaMax = cms.vdouble(2.5, 3.5), 
... 
```

**Comment by violatingcp** [comment]
Yes, 3.5-4 we had trouble gaining from the tracking. That being said, we
didn't try too hard to tune things better.

Phil

**Comment by slava77** [comment]
Some plots based on the following inputs:
/RelValTTbar_14TeV/CMSSW_9_1_0_pre1-PU25ns_90X_upgrade2023_realistic_v9_D11PU200-v1/GEN-SIM-RECO
/RelValZMM_14/CMSSW_9_1_0_pre1-PU25ns_90X_upgrade2023_realistic_v9_D12PU200-v1/GEN-SIM-RECO


ttbar PU200: some expected changes in eta>2.5
all_sign897-patvsorig-pat_ttbar14tev2023d11puwf24234p0c_patjets_slimmedjetspuppi__dqm_obj_eta

in ak4 and lower threshold variations are seen down to about eta of 2
all_sign897-patvsorig-pat_ttbar14tev2023d11puwf24234p0c_patjets_slimmedjetsak8__dqm_obj_eta
the spikes between eta 3 and 3.5 apparently have some off-scale response (e.g. below two bins in negative eta are missing), while response in the rest of the range seems OK
wfttpu200_ak4puppi_resp40to200_eta


ZMM PU200
jets in forward look somewhat OK
wfzmmpu200_ak4puppi_resp_eta_f

MET gets substantially worse
all_sign897-patvsorig-pat_zmm14tev2023d12puwf24650p0c_patmets_slimmedmetspuppi__dqm_obj_pt
wfzmmpu200_puppimet_gendiff

Phil, please confirm that the MET change is desired (supposedly as a balance of some more important improvements elsewhere).

**Comment by slava77** [comment]
@violatingcp 
in case you missed my earlier message, please check and comment on the significant MET increase in ZMM sample.

**Comment by violatingcp** [comment]
I do not understand the Gen MET plot Is this for the Z events? If so then
we cannot conclude the MET is worse or better. It should be improved in
this case after correcting for the response. Also, does this geometry
contain tracking up to eta > 4.0?

**Comment by slava77** [comment]
On 4/17/17 9:56 AM, Philip Harris wrote:
> I do not understand the Gen MET plot Is this for the Z events? If so then
> we cannot conclude the MET is worse or better. It should be improved in
> this case after correcting for the response. Also, does this geometry
> contain tracking up to eta > 4.0?

METDifference_GenMETTrue is delta(reco met, gen met).
This plot shows clearly that reco MET is now larger than GEN MET with a
significant increase of the tail ( part that's larger than 100 GeV).

Which response correction is expected on top pf slimmetMETsPuppi collection?

The sample used is
 /RelValZMM_14/CMSSW_9_1_0_pre1-PU25ns_90X_upgrade2023_realistic_v9_D12PU200-v1/GEN-SIM-RECO
It's D12 geometry with the standard tracker coverage going up to eta ~4.

**Comment by slava77** [comment]
current status, as I see it:
- for code review it would be good to check https://github.com/cms-sw/cmssw/pull/18265#pullrequestreview-32699989
- for performance: confirmation on expected/acceptable degradation in puppi MET is needed

File: CommonTools/PileupAlgos/python/Puppi_cff.py (hunk 2 above)
Comment (gpetruc) [inline CommonTools/PileupAlgos/python/Puppi_cff.py]: done

**Comment by slava77** [comment]
@cmsbuild please test

@gpetruc 
Have you checked that the physics behavior is the same?
please clarify.
Thank you.

**Comment by gpetruc** [comment]
@slava77 I checked that the change in the configuration merging the two `cms.PSet` into a single one makes no difference on the puppi MET.

**Comment by slava77** [comment]
On 4/19/17 10:16 AM, Giovanni Petrucciani wrote:
> @slava77 <https://github.com/slava77> I checked that the change in the
> configuration merging the two |cms.PSet| into a single one makes no
> difference on the puppi MET.
> 

Thank you.

So, the remaining item here is the increase in puppi MET compared to GEN
MET in ZMM events.

**Comment by violatingcp** [comment]
Hi guys,

Just following up. Attached are the response for Puppi and trackMET. As you
can see the new tune pushes the repsonse up a a little bit and the
resolution gets worse. However if you corret for the response there is no
change. This is with Z=>mumu events. The resolution change seems
potentially smaller to what Slava observes. My only guess is that the dZ
cut might be too tight for events with electrons, but I am not sure.

Phil

**Comment by slava77** [comment]
@violatingcp 
nothing was attached in the last email
Please drag and drop in the web interface.

If I read your last email correctly, there is no need to make any changes to the code now.
Please confirm.

Also, if some response correction (type-1?)  is applied on top of the plain puppi MET, the increase in MET should go away.

**Comment by slava77** [comment]
BTW, were the plots posted earlier and also the ones meant to be attached made from AOD or from remaking puppi on top of miniAOD?

We had issues with reproducibility in the past; I hope it's not the case here since it was debugged somewhat by @ahinzmann in one of the last updates to puppi reco that I recall.

**Comment by violatingcp** [comment]
Hi Slava,

Here is a link :
https://www.dropbox.com/sh/ir8toikuch95qiy/AAB8HqBRCO0NeK7k9MGDjagAa?dl=0

The point being is that the response improves with the updated sample
without an loss in response corrected resolution.  I am using the setup
directly out of the box so it should not have any reproducibility issues.

Thanks,
Phil

**Comment by slava77** [comment]
@violatingcp were your plots made directly with slimmedMETsPuppi or did you use some other MET RECO?

**Comment by violatingcp** [comment]
This is 90x re running puppi and recomputing slimmedmetspuppi it is
supposedly what slimmed mets puppi should be but i cannot guarantee

Le 21 avr. 2017 8:38 PM, "Slava Krutelyov" <notifications@github.com> a
écrit :

@violatingcp <https://github.com/violatingcp> were your plots made directly
with slimmedMETsPuppi or did you use some other MET RECO?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/cms-sw/cmssw/pull/18265#issuecomment-296269997>, or mute
the thread

**Comment by slava77** [comment]
if plots posted were made with CMSSW_9_0_0_pre4 (as it appears from the twiki) with the code mentioned in https://twiki.cern.ch/twiki/bin/view/CMS/UPGTrackerTDRStudies

I see differences between this PR and what's in that code in CommonTools/PileupAlgos/python/PhotonPuppi_cff.py: (PR -> Phil's directory)
- pt 10 -> 100
- photonId ``` egmPhotonIDs:cutBasedPhotonID-Spring15-25ns-V1-standalone-loose ``` ->  ```""```

I suppose this is what's mentioned by @gpetruc in the PR notes as something checked to be OK.
PhotonPuppi changes can affect MET significantly.

Phil, 
Were the plots posted earlier made with or without the photon changes?

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/18265	04fa5c6
- phase-2 updates: puppi tuned and \*manystripclus53X MET filters removed
- jenkins tests pass and comparisons show differences only in 2023 workflows
- local tests with PU200 samples shows that the puppi jet response changes roughly as expected (mainly in eta > 2.5). 
    - PUPPI MET in ZMM events (no intrinsic PU) shows somewhat larger than expected effects in the inclusive distributions, but the Z pT balancing plots appear to look OK (although those were made in 90X version). I checked that the pt and ID selections of PuppiPhoton mentioned in https://github.com/cms-sw/cmssw/pull/18265#issuecomment-296284442 do not change the MET plot significantly. Since the plotted slimmedMETsPuppi is type-1 corrected, it may as well be that after the corrections are updated things get better.

**Comment by violatingcp** [comment]
Hi,

the plots were made with the photon changes, but they will have 0 effect.
None of the plots we have looked at so far would be affected by this. I
would be a bit of a shame to not update the made Puppi config since the
only real change to the setup is to extend the use of the tracking out to
further eta. There is no fundamental change in the algorithm and this
should only make the performance better. Its likely the degradation, if
there is any which I seriously question, is a result of problems with
vertexing of foward tracks and not with Puppi.

Phil

**Pull request merged**